release notes/v0.30.0.md

```diff
@@ -0,0 +1,123 @@
+
+## New features
+
+### Shared memory read-only arrays between VUs ([#1739](https://github.com/loadimpact/k6/pull/1739))
+
+k6 has long had an issue with the handling of big data files with test fixtures. For example, if you have a huge `users.json` file with test users for your application:
+```js
+[
+  {"username": "user1", "password": "password1", "other": "some-long-data-....-1"},
+  {"username": "user2", "password": "password2", "other": "some-long-data-....-2"},
+  // ... ~1 million more users or more... :D
+  {"username": "user999999", "password": "password999999", "other": "some-long-data-....-999999"}
+]
+```
+
+If you just use `JSON.parse(open('users.json'))` in your script, then every VU will have a copy of the whole huge data set. Every VU in k6 is a separate JavaScript runtime, so there wasn't a thread-safe way to share data between them. Until now, that is!
+
+We've added a new built-in `SharedArray` object in the new `k6/data` module that allows VUs to share read-only data:
+```js
+import { SharedArray } from 'k6/data';
+import { sleep } from 'k6';
+import http from 'k6/http';
+
+let users = new SharedArray('someName', function () {
+    // This function will be called only once, in the first init context
+    // execution. Every other VU will just get a memory-safe read-only reference
+    // to the already opened data.
+    console.log('Loading users.json, this happens only once...');
+    // You are not restricted to JSON, you can do anything - parse a CSV or XML
+    // file, generate random data, etc. - as long as you return an array.
+    return JSON.parse(open('users.json'));
+});
+
+export let options = { vus: 10, duration: '30s' };
+export default function () {
+    let randomUserID = Math.floor(Math.random() * users.length);
+    let user = users[randomUserID]; // alternativaly, we can also use __VU and/or __ITER
+    console.log(`VU ${__VU} is running iteration ${__ITER} with user ${user.username}...`);
+    http.post('https://httpbin.test.k6.io/post', JSON.stringify(user));
+    sleep(Math.random() * 2); // or, better yet, use arribal-rate
+}
+```
+
+Notice how `Loading users.json` is logged only once, but each VU uses the `users` variable like a normal JS array. The data is read only once and we have just a single copy of the huge array in memory! Behind the scenes, k6 uses a JS [`Proxy`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Proxy) to transparently copy only the row each VU requests in `users[randomUserID]` to it. This on-demand copying is a bit inefficient, but it's still leagues better than having a copy of the huge array in every VU! And you can avoid the copying in every iteration by pinning the data used by every VU and having `let user = users[randomUserID]` in the [init context](https://k6.io/docs/using-k6/test-life-cycle)!
```

Comment:
Should we link to that memory/CPU benchmark that @MStoykov did here? It would be good to give some numbers to give an idea of the performance improvement.

Reply:
can you point me towards the benchmark? I can't seem to find it

Reply:
Found it in #1739: https://gist.github.com/MStoykov/ea62b0d16dc21261779f46db0fe8bd53

Reply:
hmm a bit contrived and not showing much benefit - I'll see if I can repeat it with a bigger array later, but in any case, not sure if we can give any benchmark results that are representative, since every case would be different.

Reply:
OK, I would've preferred to see even percentage ranges, but you're right that it will depend on the use case.